esm-loader.mjs

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import newrelic from './index.js'
+import shimmer from './lib/shimmer.js'
+import loggingModule from './lib/logger.js'
+
+const logger = loggingModule.child({ component: 'esm-loader' })
+
+/**
+ * Hook chain responsible for resolving a file URL for a given module specifier
+ *
+ * Our loader has to be the last user-supplied loader if chaining is happening,
+ * as we rely on `nextResolve` being the default Node.js resolve hook to get our URL
+ *
+ * Docs: https://nodejs.org/api/esm.html#resolvespecifier-context-nextresolve
+ *
+ * @param {string} specifier
+ *        String identifier in an import statement or import() expression
+ * @param {object} context
+ *        Metadata about the specifier, including url of the parent module and any import assertions
+ *        Optional argument that only needs to be passed when changed
+ * @param {Function} nextResolve
+ *        The Node.js default resolve hook
+ * @returns {Promise} Promise object representing the resolution of a given specifier
+ */
+export async function resolve(specifier, context, nextResolve) {
+  if (!newrelic.agent) {
```

Comment:
I think the check for `newrelic.shim` was done because it follows the patterns of our external modules.  I think checking for agent, shim or awsLambda is enough. but frankly everything we do in here I think is safe without checking anything agent properties but wasn't sure the intention when @michaelgoin wrote this into the prototype.